tests/test_api.py

```diff
@@ -430,7 +430,24 @@ def test_prewhere_conditions(self):
             'limit': 1,
             'debug': True
         })).data)
-        assert "PREWHERE project_id IN (1) AND positionCaseInsensitive(message, 'abc') != 0" in result['sql']
+        assert "PREWHERE project_id IN (1) AND positionCaseInsensitive((coalesce(search_message, message) AS message), 'abc') != 0" in result['sql']
```

Comment:
Why did these tests change, you ask?

Good question! Well, it's because before this PR, we would go through the `[['positionCaseInsensitive', ['message', "'abc'"]], '!=', 0]` condition and create an alias for `message` when evaluating the `WHERE` clause. Then, when it came time for the `PREWHERE`, we could just use the alias (i.e. `message`). 

With this change, the condition is evaluated only once (in the `PREWHERE`) and so the `PREWHERE` gets the `coalesce` bit.